warehouse/routes.py

```diff
@@ -576,6 +576,15 @@ def includeme(config):
         domain=warehouse,
     )
 
+    # PEP 740 URLs
+    config.add_route(
+        "attestations.provenance",
+        "/_/provenance/{filename}/",
```

Comment:
True...perhaps just `/provenance/{filename}`? Or even something opaque like `/provenance/{file-or-provenance-uuid}`.

(My original thinking behind `/_/` is that the PEP specifies that resolution must always go through the referenced URL, i.e., there's guarantee that it remains stable for a particular file. So we could start out on `/_/` and then move to a more permanent location, but maybe that's more effort than it's worth.)

Reply:
@ewdurbin @miketheman Any thoughts about where this route should live?

Reply:
🤔 I'm not 100% up to speed on attestations vs provenance, but why do we need an internal URL?

If requests are meant to be specific to a file, wouldn't we want to traverse to the filename in question? Something like `/project/pytest/8.3.3/files/pytest-8.3.3.tar.gz/provenance/` ?

Reply:
> 🤔 I'm not 100% up to speed on attestations vs provenance

TL;DR: "provenance" for a file is the set of one or more "attestations" for that file. It's just the JSON structure we serve to end users for their consumption/verification.

> but why do we need an internal URL?

We don't, it can be any URL (with the constraint that it needs to be something we can currently route on the Warehouse side, since provenance lives in the DB instead of object storage for now) 🙂 

> If requests are meant to be specific to a file, wouldn't we want to traverse to the filename in question? Something like `/project/pytest/8.3.3/files/pytest-8.3.3.tar.gz/provenance/` ?

I think something like this would work! I'm not super familiar with how multiple traversals work, though, so I'll need to play around with it.

Reply:
I'm curious if we anticipate any additional resources accessible in a similar way in the future. That might help establish better basis.

I don't approve of hosting them under /project/ as suggested above, since that's currently all "UI" only.

Maybe something like `/metadata/{project_name}/{release}/{filename}/provenance` works? If we ever wanted to host other metadata about a project, release, or file it seems like that would be extensible.

Reply:
Okay, I've updated this to use `/metadata/{project_name}/{release}/{filename}/provenance`, PTAL!

Reply:
A few thoughts:

* I think "metadata" is overloaded here. This isn't "metadata" in the normal sense of packaging metadata. Also, we might want to expose a new API for that metadata someday, and the presence of this route would be confusing for that (would it be at `/metadata/.../metadata`?)
* I'd like us to have the option to mount HTML-based views at this same route someday (similar to the PEP 691 simple API)
* Also similar to that API, I like that we have different top-level routes that map to different PEP-related APIs

Given that, I propose:
* we put this at `/integrity/{project_name}/{release}/{filename}/provenance`
* we add content-type negotiation for `application/vnd.pypi.integrity.v1+json`

Since we don't have an HTML-based view for PEP 740 at this time, this should 404 for now if the content-type is not set or is `text/html`.

Reply:
> Given that, I propose:
> 
> * we put this at `/integrity/{project_name}/{release}/{filename}/provenance`
> * we add content-type negotiation for `application/vnd.pypi.integrity.v1+json`
> 
> Since we don't have an HTML-based view for PEP 740 at this time, this should 404 for now if the content-type is not set or is `text/html`.

SGTM; I'll update here and you can let me know what you think!

Reply:
Done -- this now routes through `/integrity/...` and performs basic content negotiation (it has pieces in place to negotiate HTML, but currently rejects those/only accepts JSON).